src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+                "--depth=1"
+            ]);
+        var sha1 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", "HEAD"
+            ])).StandardOutput.Trim();
+        var sha2 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", $"{remoteName}/{repo2Branch}"
+            ])).StandardOutput.Trim();
+
+        var patches = await patchHandler.CreatePatches(
+            options.OutputPath,
+            sha1,
+            sha2,
+            path: null,
+            filters,
+            relativePaths: false,
+            workingDir: new NativePath(repo1),
+            applicationPath: null,
+            CancellationToken.None);
+
+        if (patches.Count == 1)
+        {
+            using FileStream fs = new(patches[0].Path, FileMode.Open, FileAccess.Read);
+            using StreamReader sr = new(fs);
+            string? line;
+            while ((line = await sr.ReadLineAsync()) != null)
+            {
+                Console.WriteLine(line);
+            }
+        }
+        else
+        {
+            Console.WriteLine("Patch was too big so it had to be split into multiple files");
+            if (UxHelpers.PromptForYesNo("Do you want to apply the patches without seeing them?"))
+            {
+                foreach (var patch in patches)
+                {
+                    await patchHandler.ApplyPatch(patch, new NativePath(repo1), removePatchAfter: true);
+                }
+            }
+            else
+            {
+                foreach (var patch in patches)
+                {
+                    Console.WriteLine($"Patch file: {patch.Path} should be applied to {patch.ApplicationPath}");
+                }
+            }
+        }
+    }
+
+    private async Task GitInitRepo(string repoPath, string branch)
+    {
+        await processManager.ExecuteGit(repoPath, [
+                "init",
+                $"--initial-branch={branch}"
+            ]);
+        await processManager.ExecuteGit(repoPath, [
+                "add", "--all"
+            ]);
+        await processManager.ExecuteGit(repoPath, [
+                "commit",
+                "-m", "Initial commit"
+            ]);
+    }
+
+    private async Task<string> PartiallyCloneVmr(string path, DiffRepo vmr, string mapping)
+    {
+        string repoPath = Path.Combine(path, "dotnet");
+        await processManager.ExecuteGit(path, [
+                "clone",
+                "--depth", "1",
+                "--filter=blob:none",
+                "--sparse",
+                vmr.Path,
+                "-b", vmr.Branch,
+                repoPath
+            ]);
+        await File.WriteAllTextAsync(Path.Combine(repoPath, GitSparseCheckoutFile), $"{VmrInfo.SourceDirName}/{mapping}");
+        await processManager.ExecuteGit(repoPath, [
+                "sparse-checkout",
+                "reapply"
+            ]);
+
+        return Path.Combine(repoPath, VmrInfo.SourceDirName, mapping);
```

Comment:
Again, if `path` is `NativePath`, then `repoPath` is `path / "dotnet"` and we can return `repoPath / VmrInfo.SourceDirName / mapping.Name`